frontend/src/__tests__/cypress/cypress/utils/oc_commands/baseCommands.ts

```diff
@@ -51,3 +51,48 @@ export const patchOpenShiftResource = (
     return result;
   });
 };
+
+/**
+ * Wait for a specific pod to become ready across all namespaces.
+ *
+ * @param podNameContains A substring to partially match against the pod name (e.g., "jupyter-nb").
+ * @returns A Cypress chainable that waits for the pod to be ready.
+ * @param timeout The amount of time to wait for the pod to become ready (default is 10 seconds, e.g., '10s', '30m').
+ * @param namespace The namespace to search for the pod. Defaults to `-A` for all namespaces, or can specify a specific namespace with `-n <namespace>`.
+ */
+export const waitForPodReady = (
+  podNameContains: string,
+  timeout = '10s',
+  namespace?: string,
+): Cypress.Chainable<CommandLineResult> => {
+  const namespaceFlag = namespace ? `-n ${namespace}` : '-A';
+  const ocCommand = `oc get pods ${namespaceFlag} --no-headers | awk '$2 ~ /^${podNameContains}/ {print $1, $2}' | xargs -tn2 oc wait --for=condition=Ready pod --timeout=${timeout} -n`;
+  cy.log(`Executing: ${ocCommand}`);
+
+  return cy.exec(ocCommand, { failOnNonZeroExit: false }).then((result: CommandLineResult) => {
+    if (result.code !== 0) {
+      throw new Error(`Pod readiness check failed: ${result.stderr}`);
+    }
+    cy.log(`Pod is ready: ${result.stdout}`);
+  });
+};
+
+/**
+ * Deletes notebooks matching a given name pattern across all namespaces.
+ *
+ * @param notebookNameContains A substring to match against the notebook name (e.g., "jupyter-nb").
+ * @returns A Cypress chainable that performs the notebook deletion process.
+ */
+export const deleteNotebook = (
+  notebookNameContains: string,
+): Cypress.Chainable<CommandLineResult> => {
+  const ocCommand = `oc get notebook -A -o custom-columns=":metadata.namespace,:metadata.name" | grep ${notebookNameContains} | xargs -I {} sh -c 'oc delete notebook -n $(echo {} | cut -d " " -f1) $(echo {} | cut -d " " -f2) --ignore-not-found'`;
+  cy.log(`Executing: ${ocCommand}`);
+
+  return cy.exec(ocCommand, { failOnNonZeroExit: false }).then((result: CommandLineResult) => {
+    if (result.code !== 0) {
+      throw new Error(`Command failed with code ${result.stderr}`);
+    }
+    cy.log(`Notebook deletion: ${result.stdout}`);
```

Comment:
Can you log when nothing has been deleted? Currently it prints like so (see image). 

![image](https://github.com/user-attachments/assets/b8a8a461-0758-465c-a66e-3bc1e7e68972)

Reply:
log added